Src/Plugins/Tests.Security/TestData/SecurePlaintextSecrets/ExpectedOutputs/SEC101_004.FacebookAppCredentials.sarif

```diff
@@ -103,11 +103,11 @@
                   "uriBaseId": "SRC_ROOT"
                 },
                 "region": {
-                  "startLine": 5,
-                  "startColumn": 14,
-                  "endLine": 5,
-                  "endColumn": 46,
-                  "charOffset": 149,
+                  "startLine": 2,
+                  "startColumn": 20,
```

Comment:
>20 [](http://example.com/codeflow?start=33&length=2)

Note several interesting and subtle points here. First, this test file changed (but the AWS did not) because in this code, there is a single shared secret and four unique ids. And so, every find uses the first secret region when it is located! Clearly, moving forward, we should add the regions for the unique ids as a related location. That can come later. It makes an important and useful point: to prevent combinations from exploding in a test file, it is helpful to use the convention of creating unique secrets but use a consistent id. That will create an issue that on double-click takes you to the specific test case. I am not suggesting changing this test file, though. We will definitely want specific tests that are dedicated to exploring the 'shared secret, unique id' and other analysis patterns. Until then, we can keep tests as they are. That is, unless a test file makes the findings explode due to all the combinations. In this case, we may prefer to update the tests to use unique secrets with shared other data, host, id, etc. Keep this in mind as you move forward with other intrafile conversions.